### DISCUSSION TITLE
Best way to rerun target after some time elapsed from last run?

### DISCUSSION THREAD

**Discussion by petrbouchal** [ropensci/targets#337] [General]
Use case: I have a remote resource which I want to update every few days, and I have no way to poll when it's changed (it cannot be collapsed into a URL). For instance, google sheets behind an auth procedure are like this.

What would be the most sensible way to trigger the target, say, every N days?

Currently, do the following:

- the command reading the data from the remote source also writes a file with the current time in it
- this target runs conditional (`tar_force()`) on the date being older than N days, which is calculated in the `_targets.R` and stored in a global variable.

This feels convoluted and I suspect I might be missing something obvious. (I am hesitant to check the file-modified date on the target file in `_targets/`, which feels more straightforward but fragile.)

Intuitively, I would like something like `tar_refresh()` that does this in one step - but perhaps there is a simple way to achieve this with the existing functions. (I know I could write a target factory to collapse the two steps into one custom target, but I am not sure how best to do the checking of last-modified/last-run time.)

**Comment by wlandau** [comment]
This came up at https://github.com/ropensci/targets/issues/131. As discussed in the thread. A completely satisfying solution would require additions to the core metadata, and I hesitate because this sort of use case is not ubiquitous. I recommend something like `tar_change(target_name, regular_command(), get_timestamp())`, where `get_timestamp()` is a custom function that rounds to the nearest week.

**Comment by petrbouchal** [comment]
Many thanks - understand you don't want to burden every step with time checking as discussed in #131. It's the combination of `tar_change()` with rounding that does the trick - thanks for that idea. And you just helped me fix my mental model of what `tar_change()` does...
